### DISCUSSION TITLE
Troubleshooting V4 Board for N64 and GB

### DISCUSSION THREAD

**Discussion by dkire13** [sanni/cartreader#477] [Need help]
Long, sorry. Trying to be thorough. 

Recently, I built the adapter boards for the Gameboy and N64. I am using them on a single slot V4 board with no battery board. Initially, they worked fine. It successfully dumped several games from each system, the only exception being the save files from Banjo Kazooie and Banjo Tooie. (eep based, it just locks up) After a few hours, however, it spontaneously stopped reading the games, outputting garbled or incorrect headers and files. GBA and N64 carts are almost never fully recognized at all, while Gameboy and Gameboy Color games have faulty headers or result in corrupted dumps. The problem is not 100% consistent, it seems to vary through power cycles or even individual attempts. Very rarely, I've managed to get functional dumps but they are not 100% identical to ones downloaded elsewhere. I'm not an expert at electronics or this cart reader, and thought reporting attempted solutions and observations might help me narrow down where to look. 

Things I've tried:

- A beefier power supply, 2.5A instead of the 1A USB port on my computer.
- Re-flashing the firmware
- Testing both with, and without, the clock generator (with appropriate firmware settings)
- Re-flowing solder on both adapter and main board, as well as the Arduino
- Testing all data pins between adapter slot and Arduino with a multimeter to ensure contact, and no shorts. 
- Testing all pins on GB adapter for contact, and no shorts.
- Ensured that the cartridge slot was receiving both 5V and 3.3V in the proper setting
- and, of course, cleaning both the cartridge and cartridge slot

Things I've noticed:

- The SNES and Genesis adapters I had made previously continue to work fine, producing identical files to ones I've dumped before. 
- The pins between the 3.3V on the arduino and the GB adapter don't fully set off my (admittedly cheap) multimeter's short detection. It does register contact, just not enough to beep. Odd, as it definitely supplies 3.3V when the unit is on. Maybe indicative of more than one problem as 5V GB and GBC games are still afflicted. Might have something to do with #369 , I have not tested that yet.
- N64 game headers are 100% consistent, a trait that none of the Gameboy titles share. The reader still reports a Gamepak error despite no data being missing from the "cartridge unknown" screen. It could be incorrect, though, I have no way of comparing to ones before it broke.
- Internal Checksum often reports success on corrupted files
- CRC32 fails to find a match every time, even on successful dumps and the SNES adapter
- Games without sram or fram on GBA work much more consistently, sometimes even completing a clean dump.
- Plugging in the fram variant of Metroid Fusion causes the entire board to power off for a split second, resetting it. Doesn't happen with any other fram game I've tested. Weird. 

Again, long post. Sorry. Any help is appreciated.

**Comment by Ancyker** [comment]
You might be hitting the issue where the Arduino/MEGA you have doesn't like running on 3.3V

> The pins between the 3.3V on the arduino and the GB adapter don't fully set off my (admittedly cheap) multimeter's short detection.

I'm not familiar with the V4 hardware but on the V5 the VCC line uses a capacitor (near the GB slot). If the V4 has this capacitor as well that can cause the continuity test to fail because the capacitor will start to charge off of the voltage supplied by the multimeter which would cause the multimeter to not see a direct short. If the voltage/current on the multimeter is slowly creeping up (or down if resistance) then this is most likely why it's not beeping. If you wait long enough it should beep once the cap is charged (if it's 470uF like the V5 it would take a long time though).

**Reply by dkire13** [reply]
Looking over the V5 instructions I can definitely see which capacitor you're talking about and can confirm the V4 does not have it on either the main board or the GB adapter. It is directly linked. Might try retro fitting one if all else fails. Should be noted that 3.3V does not seem to affect operation of the Arduino at all, it does not show any of the symptoms described in #369 . If it's a problem with power, I don't think it's voltage. Arduino says it outputs 800mA on the back, but the battery board rates itself for 1A at 3V. I don't have an 18650 cell to test, but could that be it?
